tests/oauth2/test__client.py

```diff
@@ -619,3 +619,81 @@ def test__token_endpoint_request_no_throw_with_retry(can_retry):
         assert mock_request.call_count == 4
     else:
         assert mock_request.call_count == 1
+
+
+def test_lookup_trust_boundary_request_error():
+    mock_request = mock.create_autospec(transport.Request)
+    mock_request.side_effect = Exception("Mock Request Error")
+
+    with pytest.raises(exceptions.TransportError):
+        _client.lookup_trust_boundary(mock_request, "mock_url", {})
+
+
+def test_lookup_trust_boundary():
+    test_data = [
```

Comment:
nit: unroll this into multiple test cases, otherwise it is hard to tell which test failed. Maybe use a pytest fixture? 

https://docs.pytest.org/en/stable/how-to/parametrize.html

Reply:
Fixture is a good call.
I have a plan to refactoring of each function mapping to one test with fixtures as unit test.
A whole setup scenario of a workflow as an integration test.
This is a long term gradually refactoring by a flexible pace.